### PR TITLE
File structure rework

### PR DESCRIPTION
**Pull request by IReallyHateLogins**
***
Moved property files to new folder, **copy them back to config folder**  to use them in project without tracking changes. 
***
Moved our classes to submodules, with this structure:

- **controllers** with all controllers

- **data** with all entity files and any related data classes
   - **folders** for folder representation
       - **serialzation** with all json converting tools for folders
    - **mailbox** with mailbox entity and repository
    - **user** with user entity and repository

 - **jwt** with all security configuration
    - **filters** with authentication and authorization filter
    - **userprincipial** with userprincipial and it's detail service

 - **mail** with all mail services
    - **connectors** with IMAP/SMTP connectors

***

Also fiddled with jwt security, made it so that it purely relies on UserPrincipial, put inside a comment a way to change login endpoint, I think we may add this as a setting to our properties later. @snowzhop , you should review this part.

### CODE REVIEW

File: Spring-server/src/main/java/ikpi63holding/postclient/jwt/filters/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -25,6 +28,8 @@ public JwtAuthenticationFilter(AuthenticationManager authenticationManager,
             JwtProperties jwtProperties) {
         this.authenticationManager = authenticationManager;
         this.jwtProperties = jwtProperties;
+        // We can set login endpoint here
+        //setRequiresAuthenticationRequestMatcher(new AntPathRequestMatcher("/logme", "POST"));
```
Comment (IReallyHateLogins) [inline Spring-server/src/main/java/ikpi63holding/postclient/jwt/filters/JwtAuthenticationFilter.java:32]: We can setup out endpoint like this, pulled it out of abstract class underneath.

File: Spring-server/src/main/java/ikpi63holding/postclient/jwt/filters/JwtAuthorizationFilter.java [hunk 2]
```diff
@@ -11,18 +13,19 @@
...
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
 
 public class JwtAuthorizationFilter extends BasicAuthenticationFilter {
 
     private final JwtProperties jwtProperties;
-    private final UserRepository userRepository;
+    private final UserPrincipalDetailsService principalDetailsService;
```
Comment (IReallyHateLogins) [inline Spring-server/src/main/java/ikpi63holding/postclient/jwt/filters/JwtAuthorizationFilter.java:22]: I think using UserPrincipalDetailsService here makes more sense and we can forget about userRepository on security configuration level

**Review by snowzhop — APPROVED** [review]
Nice structure

**Review by ValentinBodrov — APPROVED** [review]
This is a very clear structure

**Review by Egridnev — APPROVED** [review]

**Pull request merged**